dev-packages/electron/package.json

```diff
@@ -25,9 +26,10 @@
     "electron-store": "^2.0.0",
     "fix-path": "^2.1.0",
     "native-keymap": "^1.2.5",
+    "node-gyp": "^4.0.0",
```

Comment:
I do not understand this, also the native C part. How does it work now on the `master`?

Reply:
I can give some info and Paul can later correct any inaccuracy: the libffmpeg test on master is based on [this](https://electronjs.org/docs/development/upgrading-chromium#verify-ffmpeg-support); basically a small Electron application that attempts to play a video, encoded with a proprietary codec we want to detect (`h264`). If the video plays, it means the codec is present in `libffmpeg`, which we do not want.

The downside is that it may take a bit of effort to have this test run correctly, e.g. in downstream projects that use @theia/electron. If we do not want to break CI for those projects, we need to disable the automatic execution of this test or replace it with a test that does not require a special CI setup. 

In this PR, Paul is attempting the later: a small C program that links with `libffmpeg` and uses the library's APIs to get the list of built-in codecs. From there it's easy to check if the library contains the proprietary codecs we want to exclude. It's made in the form of a native library so that it can be compiled, on-the-fly, for the various platforms we support.

Reply:
> I can give some info and Paul can later correct any inaccuracy: the libffmpeg test on master is based on [this](https://electronjs.org/docs/development/upgrading-chromium#verify-ffmpeg-support); basically a small Electron application that attempts to play a video, encoded with a proprietary codec we want to detect (`h264`). If the video plays, it means the codec is present in `libffmpeg`, which we do not want.

Sounds good 👍

> The downside is that it may take a bit of effort to have this test run correctly, e.g. in downstream projects that use @theia/electron. If we do not want to break CI for those projects, we need to disable the automatic execution of this test or replace it with a test that does not require a special CI setup.

You can run as a pure mocha test during the test phase, downstream projects do not run the Theia tests, right? So the problem is solved.

> a small C program that links with `libffmpeg` and uses the library's APIs to get the list of built-in codecs

Writing a small C program to test this feels the most complicated solution to me. Especially using a `node-gyp@4.x` although we have `3.8.0` effective `node-gyp` dependency in the `yarn.lock`.

Reply:
> Especially using a `node-gyp@4.x` although we have `3.8.0` effective `node-gyp` dependency in the `yarn.lock`.

This is my bad, I can align both versions.

> Writing a small C program to test this feels the most complicated solution to me.

Writing C should not be considered complicated. Also this is the most robust way, as the libffmpeg library is well designed enough to expose ways to iterate over all included codecs, this  means we can detect if h264 is in, but also any other problematic codecs (to be defined).

> You can run as a pure mocha test during the test phase, downstream projects do not run the Theia tests, right? So the problem is solved.

I would have been happy too, but we wanted to have a solid case to present to the eclipse foundation, by ensuring as best as we can that everything is in order, this means always running the tests on install, which would break clients that don't have a specific CI setup.

Reply:
> You can run as a pure mocha test during the test phase, downstream projects do not run the Theia tests, right? So the problem is solved.

In a way, yes; this would fix the immediate problem for us / our repo. It would also still by default replace the library automatically for downstream projects that use @theia/electron. But it would not validate that the library was correctly replaced, without them taking action to run the test. 

> Writing a small C program to test this feels the most complicated solution to me. 

Agreed; it's more effort done this way vs the way we first did it. The gain is that other users of `@theia/electron` will have additional certainty that they are using a patent-free version of `libffmpeg`, with no additional effort on their part, validated every time they pull `@theia/electron`.